skimage/measure/_ccomp.pyx

```diff
@@ -387,14 +387,15 @@ def label_cython(input_, neighbors=None, background=None, return_num=False,
             "Connectivity below 1 or above %d is illegal."
             % ndim)
 
-    scanBG(data_p, forest_p, &shapeinfo, &bg)
-    # the data are treated as degenerated 3D arrays if needed
-    # without any performance sacrifice
-    scan3D(data_p, forest_p, &shapeinfo, &bg, connectivity)
-
+    cdef DTYPE_t conn = connectivity
     # Label output
     cdef DTYPE_t ctr
-    ctr = resolve_labels(data_p, forest_p, &shapeinfo, &bg)
+    with nogil:
```

Comment:
Is it necessary to define `nogil` if the *function* is defined `nogil`.  Also, is `nogil` necessary for `cdef` functions?

Reply:
Thanks for the find.
Apparently: https://cython.readthedocs.io/en/latest/src/userguide/external_C_code.html#declaring-a-function-as-callable-without-the-gil

Reply:
Oh sorry, I get your question. I guess I'm explicitely saying that the 3 blocks together should be taken with nogil. I'm not too sure and Cython generated code looks very ulgy.

Reply:
It looks from those docs as though a `nogil` function implicitly does not acquire the gil anywhere inside, so you should never need to combine `cdef ... nogil` and `with nogil`?

Reply:
Stefan, maybe github is misleading you, this is the function
```python
def label_cython(input_, neighbors=None, background=None, return_num=False,
                 connectivity=None):
```

Reply:
Indeed, sorry for the noise.